Added draw matches to write text and no lines as mode=3.

mpkuse opened the pull request:
When there are a lot of matches it is hard to see the correct matches. So instead of lines, I draw circles and write text (serial number of the match) to the image. Basically, I have edited your DrawInlier().

Commit:
Added draw matches to write text and no lines as mode=3. More testing with my images

Commit:
Added class GmsRobe. This class is an extention of original class to provide functionality like match3(), match2_guided() in addition to original. Returns image cords.

JiawangBian commented:
Hey mpkuse,

Thanks for your commits, and I would like to integrate it into the code. I give some suggestions before merging:

1. Formatting the name of photos, as one may not understand the which pairs should be matched.
2. Replacing "boost/timer" in cpp codes with the standard library for convenience, as many guys may not get boost library installed.
3. I have invited Konrad who implemented the original python code to check out the change. He may give suggestions below about the python scripts.

Best,
JiaWang

mpkuse commented:
Sure, I will follow your suggestion and add another commit soon.

kczarnota commented:
Hi mpkuse,

Thanks for your updates!

I have a few suggestions:

1. Can you change syntax to match Python3? The script was originally developed to support it.
2. Can you remove all unnecessary comments? I see quite a lot of code which is commented out.
3. Can you write something more about GmsRobe, which is a new functionality? Did you invented it yourself or it's a implementation of known algorithm(if yes pleas provide a link where I could read more about it).

Best,
Konrad

mpkuse commented:
What do you exactly mean by syntax to match python3? 

I will get rid of the commented code. Basically, I added the main in a separate script and class in separate so that in my projects (and for others) can simply copy the class file and follow examples from demo.py file. 

GmsRobe is my original work to extend the functionality. I will write a short readme for this. 

I think I will need a couple of days to fix this up.

Commit:
Wrote a readme for images.

wrote a readme for python usage.

removed boost timer usage. Essentially there is now no change in the
original c++ code.

I have got rid of all the unnecessary comments in code. moved the
main into demo.py. The user is supposed to just run demo.py.
It has 4 usages to choose from by keyboard input.
This demos the GmsMatcher class and GmsRobe class.

kczarnota commented:
I couldn't run your code because you use print function without parentheses. Please try to run your code with Python3 interpreter and it will tell you if there are any problems.

Take your time with changes, it's not a race :)
